### PR TITLE
Add README.

### LINKED ISSUE #2: add README
It would be nice to have a README with instructions for how to download and run this e.g. to reproduce the installation in `/g/data/ik11/spack/`

### PR DESCRIPTION
**Pull request by micaeljtoliveira**
Close #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# COSIMA Spack Configuration
+
+This repository contains the spack configuration and the spack environments used
+by COSIMA to deploy software on gadi.
+
+## Instalation instructions
```
Comment (aekiss) [inline README.md]: typo: Installation

File: README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+Clone this repository and its submodules to some appropriate location (e.g.,
+`/g/data/ik11/spack/0.20.1`):
+```bash
+$ git clone --recursive /g/data/ik11/spack/0.20.1
```
Comment (aekiss) [inline README.md]: replace with 
```
$ git clone --recursive https://github.com/COSIMA/spack-config.git /g/data/ik11/spack/0.20.1
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+First activate the spack environment
+```bash
+$ spack env activate <env>
+```
+where `<env>` by the actual name of the environment.
+
+ 2. Download the sources
+
+As the compute nodes do not have internet access, one needs to download all the
```
Comment (aekiss) [inline README.md:56]: could use `copyq` queue to get internet access

File: README.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+Here `sources` is the name of a mirror that has already been configured.
+
+ 3. Submit interactive job
+
+This should not use more than a single node. Also, make sure to add `gdata/ik11`
+and `scratch/ik11` to the storage options.
+
+ 4. Install software
+
+Once the job has started, because it starts a completly new shell session, one
```
Comment (aekiss) [inline README.md]: typo: completely

File: README.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+$ spack install 
+```
+In this case, although each individual build will use some level of parallelism,
+spack will proceed through the installation of the packages sequentially. To
+fully use parallelism one needs to tell spack to create a Makefile and use this
+to install the software:
+```bash
+$ spack env depfile -o Makefile
+$ make -j
+```
```
Comment (aekiss) [inline README.md:87]: Should something be added at the end on where the spack packages are installed and how to use them? Or will that depend too much on what is installed and/or whether they are made available as a module? (Does this set of instructions generate modules or is that a separate, optional step?)

**Comment by aekiss** [comment]
thanks for this - looks good, just a few comments

File: README.md (hunk 5 above)
Comment (aekiss) [reply README.md:87]: ok, that looks good

File: README.md (hunk 3 above)
Comment (micaeljtoliveira) [reply README.md:56]: I think it's good practice to use spack mirrors: sources are stored in a central location and easily accessible.

**Comment by aekiss** [comment]
I haven't tested these instructions but they look fine and I'm happy for you to merge this.

File: README.md (hunk 5 above)
Comment (micaeljtoliveira) [reply README.md:87]: Thanks for the suggestion!

**Comment by micaeljtoliveira** [comment]
> I haven't tested these instructions but they look fine and I'm happy for you to merge this.

I haven't tested them either in this exact form, but they follow closely the notes I wrote when I first installed spack in ik11. I've used the same notes over the last couple of days to create the new installation and everything worked fine.

File: README.md (hunk 1 above)
Comment (navidcy) [reply README.md]: @aekiss you can also use the suggestion button when reviewing and then the changes can just implemented with a click from the PR owner.

https://www.youtube.com/watch?v=rCARUlE2C0M

Comment (aekiss) [reply README.md]: awesome, thanks @navidcy, I was just thinking that would be a good feature to have!

Comment (navidcy) [reply README.md]: @github thought it first ;)

**Pull request merged**